dvc/fs/http.py

```diff
@@ -128,7 +129,7 @@ def _head(self, url):
         # Sometimes servers are configured to forbid HEAD requests
         # Context: https://github.com/iterative/dvc/issues/4131
         with self.request("GET", url, stream=True) as r:
-            if r.ok:
+            if r.ok or r.status_code == 404:
```

Comment:
Would this cause some problems when the server was actually not found (404). Some additional check are needed.